apiserver/facades/client/action/action.go

```diff
@@ -54,14 +57,31 @@ type ModelInfoService interface {
 	GetModelInfo(context.Context) (coremodel.ModelInfo, error)
 }
 
+// OperationService provides access to operations.
+type OperationService interface {
+	// Run executes an operation on various machines and units, with the provided parameters.
+	Run(ctx context.Context, params []operation.RunParams) (operation.RunResult, error)
```

Comment:
question: why are we pushing the bulk part of the call into the domain as well as the call? Not a pattern we've used before. It also makes the facade method more complicated to read.

Inside of the domain, it'll have to be run in unique transactions anyways.

This is taking push into the domain a step too far in my opinion.